scvi/model/_multivi.py

```diff
@@ -0,0 +1,709 @@
+        if n_samples > 1 and return_mean:
+            exprs = exprs.mean(0)
+        return exprs
+
+    @_doc_params(doc_differential_expression=doc_differential_expression)
+    def differential_accessibility(
+        self,
+        adata: Optional[AnnData] = None,
+        groupby: Optional[str] = None,
+        group1: Optional[Iterable[str]] = None,
+        group2: Optional[str] = None,
+        idx1: Optional[Union[Sequence[int], Sequence[bool]]] = None,
+        idx2: Optional[Union[Sequence[int], Sequence[bool]]] = None,
+        mode: Literal["vanilla", "change"] = "change",
+        delta: float = 0.05,
+        batch_size: Optional[int] = None,
+        all_stats: bool = True,
+        batch_correction: bool = False,
+        batchid1: Optional[Iterable[str]] = None,
+        batchid2: Optional[Iterable[str]] = None,
+        fdr_target: float = 0.05,
+        silent: bool = False,
+        two_sided: bool = True,
+        **kwargs,
+    ) -> pd.DataFrame:
+        r"""
+        A unified method for differential accessibility analysis.
+
+        Implements `"vanilla"` DE [Lopez18]_ and `"change"` mode DE [Boyeau19]_.
+
+        Parameters
+        ----------
+        {doc_differential_expression}
+        two_sided
+            Whether to perform a two-sided test, or a one-sided test.
+        **kwargs
+            Keyword args for :func:`scvi.utils.DifferentialComputation.get_bayes_factors`
+
+        Returns
+        -------
+        Differential accessibility DataFrame with the following columns:
+        prob_da
+            the probability of the region being differentially accessible
+        is_da_fdr
+            whether the region passes a multiple hypothesis correction procedure with the target_fdr
+            threshold
+        bayes_factor
+            Bayes Factor indicating the level of significance of the analysis
+        effect_size
+            the effect size, computed as (accessibility in population 2) - (accessibility in population 1)
+        emp_effect
+            the empirical effect, based on observed detection rates instead of the estimated accessibility
+            scores from the PeakVI model
+        est_prob1
+            the estimated probability of accessibility in population 1
+        est_prob2
+            the estimated probability of accessibility in population 2
+        emp_prob1
+            the empirical (observed) probability of accessibility in population 1
+        emp_prob2
+            the empirical (observed) probability of accessibility in population 2
+
+        """
+        adata = self._validate_anndata(adata)
+        col_names = _get_var_names_from_setup_anndata(adata)[self.n_genes :]
+        model_fn = partial(
+            self.get_accessibility_estimates, use_z_mean=False, batch_size=batch_size
+        )
+
+        # TODO check if change_fn in kwargs and raise error if so
+        def change_fn(a, b):
+            return a - b
+
+        if two_sided:
+
```

Comment:
nit: spacing is strange here, if it was for the purpose of function definitions, maybe lambdas would be more concise: (e.g. `change_fn = lambda a, b: a - b`)

Reply:
this is black formatting, though maybe a lambda is nicer?

Reply:
I think assigning lambdas to variables used to be controversial (at least PEP8 doesn't like it), but it does look nicer so I'm going with that.

Reply:
flake8 fails if you assign lambdas to a variable.